Px.Utils/Validation/ContentValidation/ContentValidator.cs

```diff
@@ -86,23 +89,20 @@ public sealed partial class ContentValidator(string filename, Encoding encoding,
                 }
             }
 
+            int lengthOfDataRows = _headingDimensionNames is not null ? GetProductOfDimensionValues(_headingDimensionNames) : 0;
+            int amountOfDataRows = _stubDimensionNames is not null ? GetProductOfDimensionValues(_stubDimensionNames) : 0;
+
             ResetFields();
 
-            return [.. feedbackItems];
+            return new ContentValidationResult([.. feedbackItems], lengthOfDataRows, amountOfDataRows);
         }
 
         /// <summary>
         /// Validates contents of Px file metadata asynchronously. Metadata syntax must be valid for this method to work properly.
         /// </summary>
-        /// <param name="entries">Array of <see cref="ValidationStructuredEntry"/> objects that represent entries of the Px file metadata</param>
-        /// <param name="customContentValidationFunctions"><see cref="ContentValidationFunctions"/> object that contains any optional additional validation functions</param>
         /// <param name="cancellationToken">Cancellation token for cancelling the validation process</param>
-        /// <returns>Array of <see cref="ValidationFeedbackItem"/> objects. Any issues found during validation will be listed here</returns>
-        public async Task<ValidationFeedbackItem[]> ValidateAsync(
-            ValidationStructuredEntry[] entries,
-            CustomContentValidationFunctions? customContentValidationFunctions = null,
-            CancellationToken cancellationToken = default
-            )
+        /// <returns><see cref="ContentValidationResult"/> object that contains the feedback gathered during the validation process.</returns>
+        public async Task<IValidationResult> ValidateAsync(CancellationToken cancellationToken = default)
```

Comment:
This should return the ContentValidationResults instead. It provides more information and omits the need for type casting when using this validator.

Reply:
Changed to return ContentValidationResult. Did same for the async functions and SyntaxValidator